src/Components/Input.tsx

```diff
@@ -1,29 +1,184 @@
 import * as React from "react";
-import { View, Text, TextInput, StyleSheet } from "react-native";
+import { StyleSheet, Text, TextInput, Animated } from "react-native";
+import Colors from "../Themes/Colors";
+import { isWeb } from "../Helpers/platform";
 
-// DEMO VERSION
-type InputProps = {
-  field: string;
-  value: string;
-  onChangeText: (value: string) => void;
+type Props = {
+  label: string;
+  onChange(res: { value: string | undefined; isValid: boolean }): void;
+  width: number | string;
+  height: number;
+  onSuccessInputFieldColor: string;
+  secureTextEntry: boolean;
+  validateValue?(value?: string): string | undefined;
+  initialValue?: string;
+  onInputFocus?(): void;
+  onInputBlur?(): void;
```

Comment:
Firstly, it has an event as a first argument

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/981cc6d51eb213943a3606c14ad75d65ea6b360f/types/react-native/index.d.ts#L1364

Secondly, this is an Input wrapper, so it should mimic Input propTypes as much as possible. We need that to make easier a drop-in replacement from native Input. So it is better to name it the same as the native one: `onInput`